runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -3237,7 +3295,11 @@ void TR_EscapeAnalysis::forceEscape(TR::Node *node, TR::Node *reason, bool force
                //candidate->setLocalAllocation(false);
                }
             else
+               {
+               if (trace())
+                  traceMsg(comp(), "  Marking candidate [%p] as referenced in forceEscape as it is immutable\n", candidate->_node);
```

Comment:
What does this mean exactly ? i.e. what does immutability have to do with marking the candidate as referenced ?

Reply:
> what does immutability have to do with marking the candidate as referenced

I was adding more tracing to make it easier to track how a candidate comes to have some of the flags set on it.  Perhaps a better way of describing this situation would be as follows:

```
traceMsg(comp(), "  Marking immutable candidate [%p] as referenced in forceEscape to allow for non-contiguous allocation, but compensating for escape at [%p]\n", candidate->_node, reason);
```

Thoughts?

Reply:
Sure that sounds better, thanks

Reply:
Updated in commit 0f1dc52e2e3a5cde23ac60ccee39d938e544f26b